Exclude GraphicsResourceManager if GL not found

Hello
Here is a build issue if GL not found : 
....
-- Could NOT find OpenGL (missing: OPENGL_INCLUDE_DIR)
...
make
...
[ 25%] Building CXX object src/backend/cuda/CMakeFiles/afcuda.dir/device_manager.cpp.o
In file included from /home/wtambellini/repos/arrayfire/src/backend/cuda/GraphicsResourceManager.cpp:18:0:
/usr/local/cuda-10.0/include/cuda_gl_interop.h:66:19: fatal error: GL/gl.h: No such file or directory
 #include <GL/gl.h>
                   ^
compilation terminated.

Here is what I see in backend/cuda/GraphicsResourceManager.cpp : 

#include <common/graphics_common.hpp>
#define __gl_h_  // FIXME Hack to avoid gl.h inclusion by cuda_gl_interop.h
#include <GraphicsResourceManager.hpp>
#include <cuda_gl_interop.h>

and the start of cuda_gl_interop.h of cuda 10.0 : 

#if !defined(__CUDA_GL_INTEROP_H__)
#define __CUDA_GL_INTEROP_H__
#include "cuda_runtime_api.h"
#if defined(__APPLE__)
#include <OpenGL/gl.h>
#else /* __APPLE__ */
#if defined(__arm__) || defined(__aarch64__)
#ifndef GL_VERSION
#error Please include the appropriate gl headers before including cuda_gl_interop.h
#endif
#else
#include <GL/gl.h>
#endif

Would is be possible to simply exclude any AF Graphics things if GL not found ?
Kind
W.

--- a/src/backend/common/CMakeLists.txt
+++ b/src/backend/common/CMakeLists.txt
@@ -76,7 +76,7 @@ target_include_directories(afcommon_interface
     ${ArrayFire_SOURCE_DIR}/src/backend
     ${ArrayFire_BINARY_DIR}
   SYSTEM INTERFACE
-    ${OPENGL_INCLUDE_DIR}
+    $<$<PLATFORM_ID:Darwin>:${OPENGL_INCLUDE_DIR}>
     ${ArrayFire_SOURCE_DIR}/extern/forge/include
     ${ArrayFire_BINARY_DIR}/extern/forge/include
   )

--- a/src/backend/cuda/CMakeLists.txt
+++ b/src/backend/cuda/CMakeLists.txt
@@ -14,6 +14,17 @@ find_cuda_helper_libs(nvrtc)
 find_cuda_helper_libs(nvrtc-builtins)
 find_cuda_helper_libs(cudnn)
 
+if(NOT OPENGL_FOUND)
+  # create a dummy gl.h header to satisfy cuda_gl_interop.h requirement
+  # all opengl functionality is made available via glad third party code
+  # that is built along with arrayfire code base.
+  set(dummy_gl_root "${PROJECT_BINARY_DIR}/include/GL")
+  if(APPLE)
+    set(dummy_gl_root "${PROJECT_BINARY_DIR}/include/OpenGL")
+  endif()
+  file(WRITE "${dummy_gl_root}/gl.h" "// Dummy file to satisy cuda_gl_interop")
+endif()
+
 get_filename_component(CUDA_LIBRARIES_PATH ${CUDA_cudart_static_LIBRARY} DIRECTORY CACHE)
 
 include(FileToString)
@@ -529,7 +540,6 @@ target_link_libraries(afcuda
     afcommon_interface
     cuda_scan_by_key
     cuda_thrust_sort_by_key
-    ${CUDA_LIBRARIES}
     ${CUDA_nvrtc_LIBRARY}
     ${CUDA_CUBLAS_LIBRARIES}
     ${CUDA_CUFFT_LIBRARIES}

--- a/src/backend/cuda/CMakeLists.txt
+++ b/src/backend/cuda/CMakeLists.txt
@@ -18,9 +18,9 @@ if(NOT OPENGL_FOUND)
   # create a dummy gl.h header to satisfy cuda_gl_interop.h requirement
   # all opengl functionality is made available via glad third party code
   # that is built along with arrayfire code base.
-  set(dummy_gl_root "${PROJECT_BINARY_DIR}/include/GL")
+  set(dummy_gl_root "${ArrayFire_BINARY_DIR}/include/GL")
   if(APPLE)
-    set(dummy_gl_root "${PROJECT_BINARY_DIR}/include/OpenGL")
+    set(dummy_gl_root "${ArrayFire_BINARY_DIR}/include/OpenGL")
   endif()
   file(WRITE "${dummy_gl_root}/gl.h" "// Dummy file to satisy cuda_gl_interop")
 endif()

--- a/src/backend/cuda/GraphicsResourceManager.cpp
+++ b/src/backend/cuda/GraphicsResourceManager.cpp
@@ -7,14 +7,11 @@
  * http://arrayfire.com/licenses/BSD-3-Clause
  ********************************************************/
 
-#if defined(OS_WIN)
-#include <windows.h>
-#endif
-
-// cuda_gl_interop.h does not include OpenGL headers for ARM
-#include <common/graphics_common.hpp>
-#define __gl_h_  // FIXME Hack to avoid gl.h inclusion by cuda_gl_interop.h
 #include <GraphicsResourceManager.hpp>
+
+#include <common/graphics_common.hpp>
+// cuda_gl_interop.h does not include OpenGL headers for ARM
+// __gl_h_ should be defined by glad.h inclusion
 #include <cuda_gl_interop.h>
 #include <err_cuda.hpp>
 #include <platform.hpp>

--- a/src/backend/cuda/device_manager.cpp
+++ b/src/backend/cuda/device_manager.cpp
@@ -26,9 +26,9 @@
 #include <cublas_v2.h> // needed for af/cuda.h
 #include <af/cuda.h>
 #include <af/version.h>
-// cuda_gl_interop.h does not include OpenGL headers for ARM
 #include <common/graphics_common.hpp>
-#define __gl_h_  // FIXME Hack to avoid gl.h inclusion by cuda_gl_interop.h
+// cuda_gl_interop.h does not include OpenGL headers for ARM
+// __gl_h_ should be defined by glad.h inclusion
 #include <cuda_gl_interop.h>
 
 #include <algorithm>

--- a/src/backend/cuda/platform.cpp
+++ b/src/backend/cuda/platform.cpp
@@ -31,10 +31,7 @@
 #include <af/cuda.h>
 #include <af/device.h>
 #include <af/version.h>
-// cuda_gl_interop.h does not include OpenGL headers for ARM
 #include <common/graphics_common.hpp>
-#define __gl_h_  // FIXME Hack to avoid gl.h inclusion by cuda_gl_interop.h
-#include <cuda_gl_interop.h>
 
 #include <algorithm>
 #include <array>

--- a/src/backend/opencl/CMakeLists.txt
+++ b/src/backend/opencl/CMakeLists.txt
@@ -447,8 +447,7 @@ if(AF_WITH_NONFREE)
 endif()
 
 if(APPLE)
-  target_link_libraries(afopencl
-    PRIVATE OpenGL::GL)
+  target_link_libraries(afopencl PRIVATE OpenGL::GL)
 endif()
 
 if(LAPACK_FOUND OR MKL_FOUND)

